lib/matplotlib/tests/test_backend_qt.py

```diff
@@ -154,11 +155,19 @@ def test_device_pixel_ratio_change():
         def set_device_pixel_ratio(ratio):
             p.return_value = ratio
 
-            # The value here doesn't matter, as we can't mock the C++ QScreen
-            # object, but can override the functional wrapper around it.
-            # Emitting this event is simply to trigger the DPI change handler
-            # in Matplotlib in the same manner that it would occur normally.
-            screen.logicalDotsPerInchChanged.emit(96)
+            current_version = tuple(int(x) for x in QtCore.qVersion().split('.', 2)[:2])
+            if current_version >= (6, 6):
+                # Emitting this event is simply to trigger the DPI change handler
+                # in Matplotlib in the same manner that it would occur normally.
+                qt_canvas.eventFilter(
```

Comment:
Is calling eventFilter() the standard way to emit a self-created event? This feels a bit hacky, which may be ok. But I'd rathe comment like this
```suggestion
                # Create an event to explicitly trigger the DPI change handler.
                # We push the event into Qt's event handling system by injecting it into
                # the eventFilter
                qt_canvas.eventFilter(
```